packages/flutter_tools/test/general.shard/plugins_test.dart

```diff
@@ -628,6 +632,78 @@ dependencies:
         },
       );
 
+      testUsingContext(
+        'Refreshing the plugin list for iOS/macOS projects invokes invalidatePodInstallOutput if the plugins changed',
+        () async {
+          // Refresh the plugin list (we have no plugins).
+          await refreshPluginsList(flutterProject, iosPlatform: true, macOSPlatform: true);
+          expect(
+            cocoaPods.capturedInvocations,
+            isEmpty,
+            reason: 'No plugins exist, so no invalidatePodInstallOutput calls expected.',
+          );
+
+          // Create an initial plugin (we previously had none).
+          createPlugin(
+            name: 'plugin-a',
+            platforms: const <String, _PluginPlatformInfo>{
+              'ios': _PluginPlatformInfo(
+                pluginClass: 'Foo',
+                dartPluginClass: 'Bar',
+                sharedDarwinSource: true,
+              ),
+            },
+          );
+          await refreshPluginsList(flutterProject, iosPlatform: true, macOSPlatform: true);
+          expect(
+            cocoaPods.capturedInvocations,
+            containsAll(<Matcher>[isA<IosProject>(), isA<MacOSProject>()]),
+            reason: 'A new plugin was added, so it should cause invalidatePodInstallOutput calls.',
+          );
+          cocoaPods.capturedInvocations.clear();
+
+          // Add a new plugin.
+          createPlugin(
+            name: 'plugin-b',
+            platforms: const <String, _PluginPlatformInfo>{
+              'ios': _PluginPlatformInfo(
+                pluginClass: 'Foo',
+                dartPluginClass: 'Bar',
+                sharedDarwinSource: true,
+              ),
+            },
+          );
+          await refreshPluginsList(flutterProject, iosPlatform: true, macOSPlatform: true);
+          expect(
+            cocoaPods.capturedInvocations,
+            containsAll(<Matcher>[isA<IosProject>(), isA<MacOSProject>()]),
+            reason: 'A new plugin was added, so it should cause invalidatePodInstallOutput calls.',
+          );
+          cocoaPods.capturedInvocations.clear();
+
+          // Do not add new plugins.
+          await refreshPluginsList(flutterProject, iosPlatform: true, macOSPlatform: true);
+          expect(
+            cocoaPods.capturedInvocations,
+            isEmpty,
+            reason: 'No plugins changed, so no updates expected',
+          );
+        },
+        overrides: <Type, Generator>{
+          CocoaPods: () => cocoaPods,
+          FileSystem: () => fs,
+          ProcessManager: () => FakeProcessManager.any(),
+          SystemClock: () => systemClock,
+          FlutterVersion: () => flutterVersion,
+          Pub: FakePubWithPrimedDeps.new,
+          // TODO(matanlurey): Remove as part of https://github.com/flutter/flutter/issues/160257.
+          // Not necessary, you can observe this bug by calling `generateLegacyPlugins: false`,
```

Comment:
Silly nitpick: consider adding links to https://github.com/flutter/flutter/issues/162399 and https://github.com/flutter/flutter/issues/163272 as we'll forget what "this bug" was :)